src/nbl/video/CVulkanLogicalDevice.cpp

```diff
@@ -1425,6 +1425,122 @@ void CVulkanLogicalDevice::createGraphicsPipelines_impl(
+    auto getVkShaderIndex = [](uint32_t index) { return index == SShaderGroupParams::SIndex::Unused ? VK_SHADER_UNUSED_KHR : index;  };
+    auto getGeneralVkRayTracingShaderGroupCreateInfo = [getVkShaderIndex](SGeneralShaderGroup group) -> VkRayTracingShaderGroupCreateInfoKHR
+    {
+        return {
+            .sType = VK_STRUCTURE_TYPE_RAY_TRACING_SHADER_GROUP_CREATE_INFO_KHR,
+            .pNext = nullptr,
+            .type = VK_RAY_TRACING_SHADER_GROUP_TYPE_GENERAL_KHR,
+            .generalShader = getVkShaderIndex(group.index),
+            .closestHitShader = VK_SHADER_UNUSED_KHR,
+            .anyHitShader = VK_SHADER_UNUSED_KHR,
+            .intersectionShader = VK_SHADER_UNUSED_KHR,
+        };
+    };
+    auto getHitVkRayTracingShaderGroupCreateInfo = [getVkShaderIndex](SHitShaderGroup group) -> VkRayTracingShaderGroupCreateInfoKHR
+    {
+        return  {
+            .sType = VK_STRUCTURE_TYPE_RAY_TRACING_SHADER_GROUP_CREATE_INFO_KHR,
+            .pNext = nullptr,
+            .type = group.intersectionShader == SShaderGroupParams::SIndex::Unused ? 
+              VK_RAY_TRACING_SHADER_GROUP_TYPE_TRIANGLES_HIT_GROUP_KHR : VK_RAY_TRACING_SHADER_GROUP_TYPE_PROCEDURAL_HIT_GROUP_KHR,
+            .generalShader = VK_SHADER_UNUSED_KHR,
+            .closestHitShader = getVkShaderIndex(group.closestHit),
+            .anyHitShader = getVkShaderIndex(group.anyHit),
+            .intersectionShader = getVkShaderIndex(group.intersectionShader),
+        };
+    };
+    for (const auto& info : createInfos)
+    {
+        initPipelineCreateInfo(outCreateInfo,info);
+        outCreateInfo->pStages = outShaderStage;
+        for (const auto& specInfo : info.shaders)
+        {
+            if (specInfo.shader)
+            {
+                *(outShaderStage++) = getVkShaderStageCreateInfoFrom(specInfo,outRequiredSubgroupSize,outSpecInfo,outSpecMapEntry,outSpecData);
+            }
+        }
+        outCreateInfo->stageCount = std::distance<decltype(outCreateInfo->pStages)>(outCreateInfo->pStages,outShaderStage);
+        assert(outCreateInfo->stageCount != 0);
+
+        const auto& shaderGroups = info.shaderGroups;
+        outCreateInfo->pGroups = outShaderGroup;
+        *(outShaderGroup++) = getGeneralVkRayTracingShaderGroupCreateInfo(shaderGroups.raygen);
+        for (const auto& shaderGroup : shaderGroups.misses)
+            *(outShaderGroup++) = getGeneralVkRayTracingShaderGroupCreateInfo(shaderGroup);
+        for (const auto& shaderGroup : shaderGroups.hits)
+            *(outShaderGroup++) = getHitVkRayTracingShaderGroupCreateInfo(shaderGroup);
+        for (const auto& shaderGroup : shaderGroups.callables)
+            *(outShaderGroup++) = getGeneralVkRayTracingShaderGroupCreateInfo(shaderGroup);
+        outCreateInfo->groupCount = 1 + shaderGroups.hits.size() + shaderGroups.misses.size() + shaderGroups.callables.size();
+        outCreateInfo->maxPipelineRayRecursionDepth = info.cached.maxRecursionDepth;
```

Comment:
need to handle the `VK_DYNAMIC_STATE_RAY_TRACING_PIPELINE_STACK_SIZE_KHR` dynamic state